### PR TITLE
Iterate in string

### LINKED ISSUE #23: New optimized write performance seems to break glamor
The recent change (in ansi.Writer) to iterating over bytes instead of runes seems to break glamor's UTF-8 processing here: https://github.com/charmbracelet/glamour/blob/master/ansi/margin.go

The first indent pipe (using ansi.Writer) breaks up UTF-8 byte sequences into single byte writes, which are misinterpreted when ingested into the second padding pipe, as they go through a byte->string->[]byte conversion.

Here's an illustration of the effect:
https://play.golang.org/p/N-caKzuAwdQ

_Originally posted by @zavislak in https://github.com/muesli/reflow/pull/10#r510675805_

### PR DESCRIPTION
**Pull request by kiyonlin**
Fix #23

### CODE REVIEW

File: ansi/writer.go [hunk 1]
```diff
@@ -12,18 +13,19 @@ type Writer struct {
...
 // Write is used to write content to the ANSI buffer.
 func (w *Writer) Write(b []byte) (int, error) {
-	for i, c := range b {
+	for _, c := range string(b) {
 		if c == '\x1B' {
 			// ANSI escape sequence
 			w.ansi = true
 			w.seqchanged = true
-			_ = w.ansiseq.WriteByte(c)
+			w.ansiseq.WriteRune(c)
```
Comment (muesli) [inline ansi/writer.go:26]: It's probably fine to ignore the error here, but if we do so, we should omit the assignment as we did before, to declare our intent. It's a bit repetitive, but I wouldn't mind proper error handling here either.

File: ansi/writer.go [hunk 2]
```diff
@@ -12,18 +13,19 @@ type Writer struct {
...
 		} else if w.ansi {
-			_ = w.ansiseq.WriteByte(c)
+			w.ansiseq.WriteRune(c)
```
Comment (muesli) [inline ansi/writer.go:28]: It's probably fine to ignore the error here, but if we do so, we should omit the assignment as we did before, to declare our intent. It's a bit repetitive, but I wouldn't mind proper error handling here either.

**Review by muesli — CHANGES_REQUESTED** [review]

File: ansi/writer.go (hunk 1 above)
Comment (kiyonlin) [reply ansi/writer.go:26]: This is because `wordwrap` does the same thing :laught:

Comment (muesli) [reply ansi/writer.go:26]: I guess I have to take the blame for that :laughing: But yeah, let's improve that!

**Comment by kiyonlin** [comment]
Could you please merge this PR first if it's ok? I'll fix the `WriteRune` stuff tomorrow.

**Comment by muesli** [comment]
Of course!

**Pull request merged**